diff --git a/dist/module.mjs b/dist/module.mjs
index 0d20b16616966904ec76211abf5a131a787130bf..e922963cd3bdfe620715fc0df37bdd37076ddf75 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -2730,22 +2730,22 @@ const nuxthub = definePreset({
   name: "nuxthub",
   async setup(options, nuxt, config) {
     const nuxtOptions = nuxt.options;
-    if (!nuxtOptions.hub?.db && !nuxtOptions.hub?.database) {
-      logger.warn("NuxtHub dedected but the database is not enabled. Using local SQLite as default database instead.");
-      return;
-    }
-    const runtimeConfig = nuxt.options.runtimeConfig;
-    const hubDb = runtimeConfig.hub.db || runtimeConfig.hub.database;
-    if (nuxtOptions.hub?.database === true) {
-      options.database ||= { type: "d1", bindingName: "DB" };
-    } else if (typeof nuxtOptions.hub?.db === "string" && typeof hubDb === "object") {
-      if (hubDb.driver === "d1") {
+    if (nuxtOptions.hub?.db || nuxtOptions.hub?.database) {
+      const runtimeConfig = nuxt.options.runtimeConfig;
+      const hubDb = runtimeConfig.hub.db || runtimeConfig.hub.database;
+      if (nuxtOptions.hub?.database === true) {
         options.database ||= { type: "d1", bindingName: "DB" };
-      } else if (hubDb.driver === "node-postgres") {
-        options.database ||= { type: "postgresql", url: hubDb.connection.url };
-      } else {
-        options.database ||= { type: hubDb.driver, ...hubDb.connection };
+      } else if (typeof nuxtOptions.hub?.db === "string" && typeof hubDb === "object") {
+        if (hubDb.driver === "d1") {
+          options.database ||= { type: "d1", bindingName: "DB" };
+        } else if (hubDb.driver === "node-postgres") {
+          options.database ||= { type: "postgresql", url: hubDb.connection.url };
+        } else {
+          options.database ||= { type: hubDb.driver, ...hubDb.connection };
+        }
       }
+    } else {
+      logger.warn("NuxtHub dedected but the database is not enabled. Using local SQLite as default database instead.");
     }
     const preset = (process.env.NITRO_PRESET || nuxt.options.nitro.preset || provider).replace(/_/g, "-");
     if (preset.includes("cloudflare")) {
