diff --git a/dist/module.mjs b/dist/module.mjs
index 0d20b16616966904ec76211abf5a131a787130bf..c44fc14a52041d654a653418ebbe4f0f5a8196b4 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -2645,6 +2645,9 @@ const node = definePreset({
     });
   },
   setupNitro(nitroConfig, { manifest }) {
+    nitroConfig.prerender ||= {};
+    nitroConfig.prerender.ignore ||= [];
+    nitroConfig.prerender.ignore.push(/^\/__nuxt_content\/.*\/sql_dump\.txt/);
     nitroConfig.publicAssets ||= [];
     nitroConfig.alias = nitroConfig.alias || {};
     nitroConfig.handlers ||= [];
@@ -2699,6 +2702,9 @@ const cloudflare = definePreset({
       logger.warn("Deploying to Cloudflare requires using D1 database, switching to D1 database with binding `DB`.");
       nitroConfig.runtimeConfig.content.database = { type: "d1", bindingName: "DB" };
     }
+    nitroConfig.prerender ||= {};
+    nitroConfig.prerender.ignore ||= [];
+    nitroConfig.prerender.ignore.push(/^\/__nuxt_content\/.*\/sql_dump\.txt/);
     nitroConfig.publicAssets ||= [];
     nitroConfig.alias = nitroConfig.alias || {};
     nitroConfig.handlers ||= [];
