diff --git a/dist/index.mjs b/dist/index.mjs
index e8ba90c0d8410d3718875007c469e44ee5b7cd69..b4d218457a186550b03b993891e2caae8d59b8c2 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1601,10 +1601,20 @@ function SSRStylesPlugin(nuxt) {
       if (!chunk) {
         continue;
       }
+      chunk.css &&= [];
       if (chunk.isEntry && chunk.src) {
         entryIds.add(chunk.src);
-      } else {
-        chunk.css &&= [];
+      }
+    }
+    // Clear entry CSS when shouldInline is true, or when function returns true for entry
+    const shouldInline = nuxt.options.features.inlineStyles;
+    for (const chunk of Object.values(manifest)) {
+      if (chunk.isEntry && chunk.src) {
+        entryIds.add(chunk.src);
+        const shouldClearCSS = shouldInline === true || (typeof shouldInline === "function" && shouldInline(entry));
+        if (shouldClearCSS) {
+          chunk.css &&= [];
+        }
       }
     }
     nitro.options.virtual["#internal/nuxt/entry-ids.mjs"] = () => `export default ${JSON.stringify(Array.from(entryIds))}`;
