diff --git a/dist/runtime/config.d.ts b/dist/runtime/config.d.ts
index 5e776f79d31a82d150124fa3dbc2be3f25d8963b..961e7b3a68eaed505ab46dec5113ce9f94ea9298 100644
--- a/dist/runtime/config.d.ts
+++ b/dist/runtime/config.d.ts
@@ -1,6 +1,6 @@
 import type { BetterAuthOptions, BetterAuthPlugin } from 'better-auth';
 import type { BetterAuthClientOptions } from 'better-auth/client';
-import type { CasingOption } from '../schema-generator.js';
+import type { Casing as CasingOption } from 'drizzle-orm/utils';
 import type { ServerAuthContext } from './types/augment.js';
 import { createAuthClient } from 'better-auth/vue';
 export type { ServerAuthContext };
