diff --git a/dist/pages/runtime/page.js b/dist/pages/runtime/page.js
index 5784808e789939a852743be7d124e65f66409269..133cd43751533dde331e950fb070037decaad498 100644
--- a/dist/pages/runtime/page.js
+++ b/dist/pages/runtime/page.js
@@ -59,12 +59,12 @@ export default defineComponent({
     }
     let pageLoadingEndHookAlreadyCalled = false;
     if (import.meta.client) {
+      // Ensure hydration completes if unmounted before Suspense resolves (e.g., layout change)
+      onBeforeUnmount(done);
       const unsub = useRouter().beforeResolve(() => {
         pageLoadingEndHookAlreadyCalled = false;
       });
-      onBeforeUnmount(() => {
-        unsub();
-      });
+      onBeforeUnmount(unsub);
     }
     return () => {
       return h(RouterView, { name: props.name, route: props.route, ...attrs }, {
