diff --git a/dist/module.mjs b/dist/module.mjs
index 8702a2e5e268b3f7ed049c2b5cc3dd42d4bb95f2..06b92c99dc09e94310f6d64a68bc82a1651de494 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -5,7 +5,8 @@ import { consola as consola$1 } from 'consola';
 import { isAbsolute, join, relative, dirname } from 'pathe';
 import { defu } from 'defu';
 import { toRouteMatcher, createRouter } from 'radix3';
-import { generateDrizzleSchema as generateDrizzleSchema$1 } from 'auth/api';
+import { generateSchema } from 'auth/api';
+import { drizzleAdapter as drizzleRelationsV2Adapter } from '@better-auth/drizzle-adapter/relations-v2';
 import { randomBytes } from 'node:crypto';
 import { isCI, isTest } from 'std-env';
 export { defineClientAuth, defineServerAuth } from '../dist/runtime/config.js';
@@ -285,16 +286,13 @@ async function generateDrizzleSchema(authOptions, dialect, schemaOptions) {
       }
     }
   };
-  const adapter = {
-    id: "drizzle",
-    options: {
-      provider,
-      camelCase: schemaOptions?.casing !== "snake_case",
-      schemaName: schemaOptions?.schemaName,
-      adapterConfig: { usePlural: schemaOptions?.usePlural ?? false }
-    }
-  };
-  const result = await generateDrizzleSchema$1({
+  const adapter = drizzleRelationsV2Adapter({}, {
+    provider,
+    camelCase: schemaOptions?.casing !== "snake_case",
+    schemaName: schemaOptions?.schemaName,
+    usePlural: schemaOptions?.usePlural ?? false
+  })(options);
+  const result = await generateSchema({
     adapter,
     options
   });
