diff --git a/dist/runtime-core.cjs.js b/dist/runtime-core.cjs.js
index 08899205dfe52109b460e9b8fc9847eb5b3241f7..b71938419a6479e152acab42af9cac80c7aa3057 100644
--- a/dist/runtime-core.cjs.js
+++ b/dist/runtime-core.cjs.js
@@ -5513,6 +5513,12 @@ function baseCreateRenderer(options, createHydrationFns) {
       optimized = false;
       n2.dynamicChildren = null;
     }
+    if (n2.dynamicChildren && n1 && n1.dynamicChildren && n1.dynamicChildren.hasOnce) {
+      if (n2.dynamicChildren === shared.EMPTY_ARR) {
+        n2.dynamicChildren = [];
+      }
+      n2.dynamicChildren.hasOnce = true;
+    }
     const { type, ref, shapeFlag } = n2;
     switch (type) {
       case Text:
@@ -6720,7 +6726,11 @@ function baseCreateRenderer(options, createHydrationFns) {
       reactivity.resetTracking();
     }
     if (cacheIndex != null) {
-      parentComponent.renderCache[cacheIndex] = void 0;
+      const cache = (vnode.ctx || parentComponent).renderCache;
+      const cached = cache[cacheIndex];
+      if (cached && (cached.component ? cached.component === vnode.component : cached.el === vnode.el)) {
+        cache[cacheIndex] = void 0;
+      }
     }
     if (shapeFlag & 256) {
       parentComponent.ctx.deactivate(vnode);
@@ -7913,7 +7923,8 @@ function cloneVNode(vnode, extraProps, mergeRef = false, cloneTransition = false
     el: vnode.el,
     anchor: vnode.anchor,
     ctx: vnode.ctx,
-    ce: vnode.ce
+    ce: vnode.ce,
+    cacheIndex: vnode.cacheIndex
   };
   if (transition && cloneTransition) {
     setTransitionHooks(
diff --git a/dist/runtime-core.cjs.prod.js b/dist/runtime-core.cjs.prod.js
index 8b2db9c0e26d3d18bf71bbf0922b61c555018708..6ac02db5610257702b77e1ee3ada8cbbac3fa54d 100644
--- a/dist/runtime-core.cjs.prod.js
+++ b/dist/runtime-core.cjs.prod.js
@@ -4138,6 +4138,12 @@ function baseCreateRenderer(options, createHydrationFns) {
       optimized = false;
       n2.dynamicChildren = null;
     }
+    if (n2.dynamicChildren && n1 && n1.dynamicChildren && n1.dynamicChildren.hasOnce) {
+      if (n2.dynamicChildren === shared.EMPTY_ARR) {
+        n2.dynamicChildren = [];
+      }
+      n2.dynamicChildren.hasOnce = true;
+    }
     const { type, ref, shapeFlag } = n2;
     switch (type) {
       case Text:
@@ -5228,7 +5234,11 @@ function baseCreateRenderer(options, createHydrationFns) {
       reactivity.resetTracking();
     }
     if (cacheIndex != null) {
-      parentComponent.renderCache[cacheIndex] = void 0;
+      const cache = (vnode.ctx || parentComponent).renderCache;
+      const cached = cache[cacheIndex];
+      if (cached && (cached.component ? cached.component === vnode.component : cached.el === vnode.el)) {
+        cache[cacheIndex] = void 0;
+      }
     }
     if (shapeFlag & 256) {
       parentComponent.ctx.deactivate(vnode);
@@ -6330,7 +6340,8 @@ function cloneVNode(vnode, extraProps, mergeRef = false, cloneTransition = false
     el: vnode.el,
     anchor: vnode.anchor,
     ctx: vnode.ctx,
-    ce: vnode.ce
+    ce: vnode.ce,
+    cacheIndex: vnode.cacheIndex
   };
   if (transition && cloneTransition) {
     setTransitionHooks(
diff --git a/dist/runtime-core.esm-bundler.js b/dist/runtime-core.esm-bundler.js
index f0905f8afe50421feed03b208b19afb26e4bbb73..5acaac3264ce0984e6d03b81e6a49dbc781a1683 100644
--- a/dist/runtime-core.esm-bundler.js
+++ b/dist/runtime-core.esm-bundler.js
@@ -5563,6 +5563,12 @@ function baseCreateRenderer(options, createHydrationFns) {
       optimized = false;
       n2.dynamicChildren = null;
     }
+    if (n2.dynamicChildren && n1 && n1.dynamicChildren && n1.dynamicChildren.hasOnce) {
+      if (n2.dynamicChildren === EMPTY_ARR) {
+        n2.dynamicChildren = [];
+      }
+      n2.dynamicChildren.hasOnce = true;
+    }
     const { type, ref, shapeFlag } = n2;
     switch (type) {
       case Text:
@@ -6781,7 +6787,11 @@ function baseCreateRenderer(options, createHydrationFns) {
       resetTracking();
     }
     if (cacheIndex != null) {
-      parentComponent.renderCache[cacheIndex] = void 0;
+      const cache = (vnode.ctx || parentComponent).renderCache;
+      const cached = cache[cacheIndex];
+      if (cached && (cached.component ? cached.component === vnode.component : cached.el === vnode.el)) {
+        cache[cacheIndex] = void 0;
+      }
     }
     if (shapeFlag & 256) {
       parentComponent.ctx.deactivate(vnode);
@@ -7974,7 +7984,8 @@ function cloneVNode(vnode, extraProps, mergeRef = false, cloneTransition = false
     el: vnode.el,
     anchor: vnode.anchor,
     ctx: vnode.ctx,
-    ce: vnode.ce
+    ce: vnode.ce,
+    cacheIndex: vnode.cacheIndex
   };
   if (transition && cloneTransition) {
     setTransitionHooks(
