diff --git a/dist/module.mjs b/dist/module.mjs
index 134fd3c5a270edbe7974e64713e0f91b0c6a773b..040a43fa4c6c8f060b2fbf8b365f428237c803eb 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -1,4 +1,4 @@
-import { readFile, writeFile, mkdir, copyFile } from 'node:fs/promises';
+import { readFile, writeFile, mkdir, copyFile, stat } from 'node:fs/promises';
 import { logger, createResolver, addServerPlugin, getLayerDirectories, updateTemplates, addTemplate, addTypeTemplate, addServerHandler, addServerImports, addImportsDir, defineNuxtModule } from '@nuxt/kit';
 import { join, resolve as resolve$1, relative } from 'pathe';
 import { defu } from 'defu';
@@ -220,7 +220,7 @@ async function resolveDatabaseConfig(nuxt, hub) {
         }
         break;
       }
-      if (hub.hosting.includes("cloudflare") && !nuxt.options.dev) {
+      if (hub.hosting.includes("cloudflare") && !nuxt.options.dev && !nuxt.options._prepare) {
         config.driver = "d1";
         break;
       }
@@ -368,47 +368,32 @@ async function generateDatabaseSchema(nuxt, hub) {
     getContents: () => `${schemaPaths.map((path) => `export * from '${path}'`).join("\n")}`,
     write: true
   });
-  if (!nuxt.options._prepare) {
-    nuxt.hooks.hookOnce("app:templatesGenerated", async () => {
-      await buildDatabaseSchema(nuxt.options.buildDir, { relativeDir: nuxt.options.rootDir });
-      const physicalDbDir = join(nuxt.options.rootDir, "node_modules", "@nuxthub", "db");
-      await mkdir(physicalDbDir, { recursive: true });
+  nuxt.hooks.hookOnce("app:templatesGenerated", async () => {
+    await buildDatabaseSchema(nuxt.options.buildDir, { relativeDir: nuxt.options.rootDir });
+    const physicalDbDir = join(nuxt.options.rootDir, "node_modules", "@nuxthub", "db");
+    await mkdir(physicalDbDir, { recursive: true });
+    try {
+      await copyFile(join(nuxt.options.buildDir, "hub/db/schema.mjs"), join(physicalDbDir, "schema.mjs"));
+      const buildDirSource = join(nuxt.options.buildDir, "hub/db/schema.d.mts");
+      const nuxtDirSource = join(nuxt.options.rootDir, ".nuxt/hub/db/schema.d.mts");
+      let schemaTypes;
       try {
-        await copyFile(join(nuxt.options.buildDir, "hub/db/schema.mjs"), join(physicalDbDir, "schema.mjs"));
-        const schemaDtsSource = join(nuxt.options.buildDir, "hub/db/schema.d.mts");
+        schemaTypes = await readFile(buildDirSource, "utf-8");
+      } catch {
         try {
-          const schemaTypes = await readFile(schemaDtsSource, "utf-8");
-          await writeFile(join(physicalDbDir, "schema.d.mts"), schemaTypes);
+          schemaTypes = await readFile(nuxtDirSource, "utf-8");
         } catch {
-          await writeFile(
-            join(physicalDbDir, "schema.d.mts"),
-            `export * from './schema.mjs'`
-          );
         }
-        const packageJson = {
-          name: "@nuxthub/db",
-          version: "0.0.0",
-          type: "module",
-          exports: {
-            ".": {
-              types: "./db.d.ts",
-              default: "./db.mjs"
-            },
-            "./schema": {
-              types: "./schema.d.mts",
-              default: "./schema.mjs"
-            }
-          }
-        };
-        await writeFile(
-          join(physicalDbDir, "package.json"),
-          JSON.stringify(packageJson, null, 2)
-        );
-      } catch (error) {
-        log$3.warn(`Failed to copy schema to node_modules/.hub/: ${error}`);
       }
-    });
-  }
+      if (schemaTypes && schemaTypes.length > 50) {
+        await writeFile(join(physicalDbDir, "schema.d.mts"), schemaTypes);
+      } else if (!nuxt.options.test) {
+        await writeFile(join(physicalDbDir, "schema.d.mts"), `export * from './schema.mjs'`);
+      }
+    } catch (error) {
+      log$3.warn(`Failed to copy schema to node_modules/.hub/: ${error}`);
+    }
+  });
   nuxt.options.alias ||= {};
   addTypeTemplate({
     filename: "hub/db/schema.d.ts",
@@ -493,7 +478,7 @@ const apiToken = ${JSON.stringify(connection.apiToken)}
 async function d1HttpDriver(sql, params, method) {
   if (method === 'values') method = 'all'
 
-  const { errors, success, result } = await $fetch(\`https://api.cloudflare.com/client/v4/accounts/\${accountId}/d1/db/\${databaseId}/raw\`, {
+  const { errors, success, result } = await $fetch(\`https://api.cloudflare.com/client/v4/accounts/\${accountId}/d1/database/\${databaseId}/raw\`, {
     method: 'POST',
     headers: {
       Authorization: \`Bearer \${apiToken}\`,
@@ -599,6 +584,26 @@ export const db: ReturnType<typeof drizzleCore<typeof schema>>
     join(physicalDbDir, "db.d.ts"),
     physicalDbTypes
   );
+  const packageJson = {
+    name: "@nuxthub/db",
+    version: "0.0.0",
+    type: "module",
+    exports: {
+      ".": { types: "./db.d.ts", default: "./db.mjs" },
+      "./schema": { types: "./schema.d.mts", default: "./schema.mjs" }
+    }
+  };
+  try {
+    await writeFile(join(physicalDbDir, "package.json"), JSON.stringify(packageJson, null, 2));
+    const schemaPath = join(physicalDbDir, "schema.mjs");
+    const schemaDtsPath = join(physicalDbDir, "schema.d.mts");
+    const schemaExists = await stat(schemaPath).then((s) => s.size > 20).catch(() => false);
+    const schemaDtsExists = await stat(schemaDtsPath).then((s) => s.size > 20).catch(() => false);
+    if (!schemaExists) await writeFile(schemaPath, "export {}");
+    if (!schemaDtsExists) await writeFile(schemaDtsPath, "export {}");
+  } catch (error) {
+    throw new Error(`Failed to create @nuxthub/db package files: ${error.message}`);
+  }
   nuxt.options.alias["hub:db"] = "@nuxthub/db";
   addServerImports({ name: "db", from: "@nuxthub/db", meta: { description: `The ${driver} database client.` } });
   addServerImports({ name: "schema", from: "@nuxthub/db", meta: { description: `The database schema object` } });
