diff --git a/dist/db/lib/client.mjs b/dist/db/lib/client.mjs
index d74fc6948b817e41a0e05bca4e6500eecacabb15..0b3c6db45da02c168eea7f07a66c11b4023a0cfe 100644
--- a/dist/db/lib/client.mjs
+++ b/dist/db/lib/client.mjs
@@ -3,7 +3,7 @@ import { join } from "pathe";
 async function createD1HttpClient(accountId, databaseId, apiToken, casing) {
   const d1HttpDriver = async (sql, params, method) => {
     if (method === "values") method = "all";
-    const response = await fetch(`https://api.cloudflare.com/client/v4/accounts/${accountId}/d1/db/${databaseId}/raw`, {
+    const response = await fetch(`https://api.cloudflare.com/client/v4/accounts/${accountId}/d1/database/${databaseId}/raw`, {
       method: "POST",
       headers: { Authorization: `Bearer ${apiToken}`, "Content-Type": "application/json" },
       body: JSON.stringify({ sql, params })
diff --git a/dist/module.mjs b/dist/module.mjs
index 1ef9c1437928b792369a168489dbf4ac6842e334..29b9a507f0bda7e084d5585cf4dc44a73459fcf4 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -314,7 +314,7 @@ async function setupDatabase(nuxt, hub, deps) {
       const dbBinding = d1Databases.find((db) => db.binding === "DB");
       if (dbBinding) {
         dbBinding.migrations_table ||= "_hub_migrations";
-        dbBinding.migrations_dir ||= ".output/server/db/migrations/";
+        dbBinding.migrations_dir ||= "db/migrations/sqlite/";
       }
     });
   }
