diff --git a/dist/module.mjs b/dist/module.mjs
index 1ef9c1437928b792369a168489dbf4ac6842e334..5a121ec6ec12fca48f12888123ffc5cc4cb6c35e 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -234,11 +234,12 @@ async function resolveDatabaseConfig(nuxt, hub) {
       break;
     }
     case "postgresql": {
-      if (hub.hosting.includes("cloudflare") && config.connection?.hyperdriveId && !config.driver) {
-        config.driver = "postgres-js";
+      config.connection = defu(config.connection, { url: process.env.POSTGRES_URL || process.env.POSTGRESQL_URL || process.env.DATABASE_URL || "" });
+      if (hub.hosting.includes("cloudflare") && config.connection?.hyperdriveId && !config.connection.url && (!config.driver || config.driver === "postgres-js")) {
+        config.driver ||= "postgres-js";
+        config.applyMigrationsDuringBuild = false;
         break;
       }
-      config.connection = defu(config.connection, { url: process.env.POSTGRES_URL || process.env.POSTGRESQL_URL || process.env.DATABASE_URL || "" });
       if (config.applyMigrationsDuringBuild && config.driver && ["neon-http", "postgres-js"].includes(config.driver) && !config.connection.url) {
         throw new Error(`\`${config.driver}\` driver requires \`DATABASE_URL\`, \`POSTGRES_URL\`, or \`POSTGRESQL_URL\` environment variable when \`applyMigrationsDuringBuild\` is enabled`);
       }
@@ -252,12 +253,13 @@ async function resolveDatabaseConfig(nuxt, hub) {
       break;
     }
     case "mysql": {
-      if (hub.hosting.includes("cloudflare") && config.connection?.hyperdriveId && !config.driver) {
-        config.driver = "mysql2";
+      config.connection = defu(config.connection, { uri: process.env.MYSQL_URL || process.env.DATABASE_URL || "" });
+      if (hub.hosting.includes("cloudflare") && config.connection?.hyperdriveId && !config.connection.uri) {
+        config.driver ||= "mysql2";
+        config.applyMigrationsDuringBuild = false;
         break;
       }
       config.driver ||= "mysql2";
-      config.connection = defu(config.connection, { uri: process.env.MYSQL_URL || process.env.DATABASE_URL || "" });
       if (config.applyMigrationsDuringBuild && !config.connection.uri) {
         throw new Error("MySQL requires DATABASE_URL or MYSQL_URL environment variable when `applyMigrationsDuringBuild` is enabled");
       }
