diff --git a/dist/runtime/virtual/_runtime_warn.mjs b/dist/runtime/virtual/_runtime_warn.mjs
index eb9a1452af48eed559258fdb41a1e37bab5275a5..147eafee1723d981c2262dff7809fe66c100953f 100644
--- a/dist/runtime/virtual/_runtime_warn.mjs
+++ b/dist/runtime/virtual/_runtime_warn.mjs
@@ -1,5 +1,5 @@
-import consola from "consola";
-import { isTest } from "std-env";
+const env = globalThis.process?.env;
+const isTest = env?.NODE_ENV === "test" || !!env?.TEST;
 if (!isTest) {
-	consola.warn("Nitro runtime imports detected without a builder or Nitro plugin. A stub implementation will be used.");
+	console.warn("Nitro runtime imports detected without a builder or Nitro plugin. A stub implementation will be used.");
 }
