diff --git a/dist/index.mjs b/dist/index.mjs
index b85e450d8e6961c0c4fa72827fea3b4889981071..327b2bfea6ad8d6a7572ef279b092ddf3d5a0063 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -14,7 +14,7 @@ import onChange from 'on-change';
 import { colors } from 'consola/utils';
 import { resolveCompatibilityDatesFromEnv, formatDate } from 'compatx';
 import escapeRE from 'escape-string-regexp';
-import { withTrailingSlash as withTrailingSlash$1, parseURL, parseQuery, joinURL, withLeadingSlash, encodePath, withoutLeadingSlash } from 'ufo';
+import { withTrailingSlash as withTrailingSlash$1, parseURL, parseQuery, joinURL, withLeadingSlash, withoutLeadingSlash } from 'ufo';
 import { ImpoundPlugin } from 'impound';
 import defu$1, { defu } from 'defu';
 import { satisfies, coerce, gt } from 'semver';
@@ -483,7 +483,7 @@ function getRouteMeta(contents, absolutePath, extraExtractionKeys = /* @__PURE__
   extractCache[absolutePath] = extractedData;
   return klona(extractedData);
 }
-const COLON_RE = /:/g;
+const ESCAPE_CHARS_RE = /[\\:]/g;
 function getRoutePath(tokens, hasSucceedingSegment = false) {
   return tokens.reduce((path, token) => {
     switch (token.type) {
@@ -497,7 +497,7 @@ function getRoutePath(tokens, hasSucceedingSegment = false) {
         return path;
       case SegmentTokenType.static:
       default:
-        return path + encodePath(token.value).replace(COLON_RE, "\\:");
+        return path + token.value.replace(ESCAPE_CHARS_RE, "\\$&");
     }
   }, "/");
 }
