diff --git a/dist/module.mjs b/dist/module.mjs
index 4d59df6ff1e84832b835d220175acf62fe68a7b4..9bfce3f4492c9719863ffaf6ab4ccf0a38cd1615 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -1,6 +1,6 @@
 import { existsSync } from 'node:fs';
 import { mkdir, writeFile } from 'node:fs/promises';
-import { defineNuxtModule, createResolver, hasNuxtModule, addTemplate, addTypeTemplate, updateTemplates, addServerImportsDir, addServerScanDir, addServerHandler, addImportsDir, addPlugin, addComponentsDir, extendPages } from '@nuxt/kit';
+import { defineNuxtModule, createResolver, hasNuxtModule, addTemplate, addTypeTemplate, updateTemplates, addServerImports, addServerImportsDir, addServerScanDir, addServerHandler, addImportsDir, addPlugin, addComponentsDir, extendPages } from '@nuxt/kit';
 import { consola as consola$1 } from 'consola';
 import { defu } from 'defu';
 import { join } from 'pathe';
@@ -162,7 +162,9 @@ function getMysqlType(type, fieldName) {
 }
 async function loadUserAuthConfig(configPath, throwOnError = false) {
   const { createJiti } = await import('jiti');
-  const jiti = createJiti(import.meta.url, { interopDefault: true });
+  const { defineServerAuth } = await import('./runtime/config.js');
+  const jiti = createJiti(import.meta.url, { interopDefault: true, moduleCache: false });
+  globalThis.defineServerAuth = defineServerAuth;
   try {
     const mod = await jiti.import(configPath);
     const configFn = typeof mod === "object" && mod !== null && "default" in mod ? mod.default : mod;
@@ -370,6 +372,7 @@ export {}
       }
     });
     addServerImportsDir(resolver.resolve("./runtime/server/utils"));
+    addServerImports([{ name: "defineServerAuth", from: resolver.resolve("./runtime/config") }]);
     addServerScanDir(resolver.resolve("./runtime/server/middleware"));
     addServerHandler({ route: "/api/auth/**", handler: resolver.resolve("./runtime/server/api/auth/[...all]") });
     addImportsDir(resolver.resolve("./runtime/app/composables"));
