diff --git a/dist/module.mjs b/dist/module.mjs
index 134fd3c5a270edbe7974e64713e0f91b0c6a773b..de979ee6e3bc8fa7d81633976056eb16cecb390c 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -368,9 +368,11 @@ async function generateDatabaseSchema(nuxt, hub) {
     getContents: () => `${schemaPaths.map((path) => `export * from '${path}'`).join("\n")}`,
     write: true
   });
+  nuxt.hooks.hookOnce("app:templatesGenerated", async () => {
+    await buildDatabaseSchema(nuxt.options.buildDir, { relativeDir: nuxt.options.rootDir });
+  });
   if (!nuxt.options._prepare) {
     nuxt.hooks.hookOnce("app:templatesGenerated", async () => {
-      await buildDatabaseSchema(nuxt.options.buildDir, { relativeDir: nuxt.options.rootDir });
       const physicalDbDir = join(nuxt.options.rootDir, "node_modules", "@nuxthub", "db");
       await mkdir(physicalDbDir, { recursive: true });
       try {
@@ -380,30 +382,18 @@ async function generateDatabaseSchema(nuxt, hub) {
           const schemaTypes = await readFile(schemaDtsSource, "utf-8");
           await writeFile(join(physicalDbDir, "schema.d.mts"), schemaTypes);
         } catch {
-          await writeFile(
-            join(physicalDbDir, "schema.d.mts"),
-            `export * from './schema.mjs'`
-          );
+          await writeFile(join(physicalDbDir, "schema.d.mts"), `export * from './schema.mjs'`);
         }
         const packageJson = {
           name: "@nuxthub/db",
           version: "0.0.0",
           type: "module",
           exports: {
-            ".": {
-              types: "./db.d.ts",
-              default: "./db.mjs"
-            },
-            "./schema": {
-              types: "./schema.d.mts",
-              default: "./schema.mjs"
-            }
+            ".": { types: "./db.d.ts", default: "./db.mjs" },
+            "./schema": { types: "./schema.d.mts", default: "./schema.mjs" }
           }
         };
-        await writeFile(
-          join(physicalDbDir, "package.json"),
-          JSON.stringify(packageJson, null, 2)
-        );
+        await writeFile(join(physicalDbDir, "package.json"), JSON.stringify(packageJson, null, 2));
       } catch (error) {
         log$3.warn(`Failed to copy schema to node_modules/.hub/: ${error}`);
       }
