diff --git a/nodejs/crypto.mjs b/nodejs/crypto.mjs
index 314f7fb2fdf084a98b7dd6d07fed8259467cf411..3e138a0538b82a731b8ca1e449314615e0a47fef 100644
--- a/nodejs/crypto.mjs
+++ b/nodejs/crypto.mjs
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 import { parentPort } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.mjs';
 import { CryptoUtils } from './crypto-wasm/index.js';
 
diff --git a/nodejs/index.mjs b/nodejs/index.mjs
index 5ef17efd9038b885582ecb6c97a474a7b0ca802b..7fdad4a6f41fece8622e8135513ca6ccd276445a 100644
--- a/nodejs/index.mjs
+++ b/nodejs/index.mjs
@@ -1,6 +1,6 @@
 import { webcrypto } from 'node:crypto';
 import { Worker } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.min.mjs';
 import { Address, CryptoUtils, Transaction } from './main-wasm/index.js';
 import { clientFactory } from '../launcher/node/client-proxy.mjs';
diff --git a/nodejs/worker.mjs b/nodejs/worker.mjs
index 18fb7cce7ce1b99d0a261886d748f814a23d1221..4231f6b33f11cf70e539e8e66a4babd420819275 100644
--- a/nodejs/worker.mjs
+++ b/nodejs/worker.mjs
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 import { parentPort } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.mjs';
 import websocket from 'websocket';
 import { Client } from './worker-wasm/index.js';
